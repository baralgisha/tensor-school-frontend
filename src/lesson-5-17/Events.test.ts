import {createButton, createList, createLink} from './Events';

describe('Урок 5.17 - События', () => {
   describe('Создание и удаление кнопки по клику', () => {
      afterEach(() => {
         document.body.innerHTML = '';
      });

      it('Создана кнопка в body "Удали меня"', () => {
         createButton();
         expect(document.body).toMatchSnapshot();
      });

      it('Клик по кнопке "Удали меня"', () => {
         createButton();
         document.body.getElementsByTagName('button')[0].click();
         expect(document.body).toMatchSnapshot();
      });
   });

   describe('Создание списка и запись атрибута по наведению', () => {
      afterEach(() => {
         document.body.innerHTML = '';
      });

      it('Создан список в body из элементов массива', () => {
         createList(['нулевой элемент', 'первый элемент', 'второй элемент']);
         expect(document.body).toMatchSnapshot();
      });

      it('Наведение мыши на элементы списка', () => {
         const arr = ['нулевой элемент', 'первый элемент', 'второй элемент'];
         createList(arr);

         function triggerEvent(el: HTMLElement, type: string): void {
            if ('createEvent' in document) {
               var e = document.createEvent('HTMLEvents');
               e.initEvent(type, false, true);
               el.dispatchEvent(e);
            }
         }

         for (let i = 0; i < arr.length; i++) {
            triggerEvent(document.body.getElementsByTagName('li')[i], 'mouseover');
         }

         expect(document.body).toMatchSnapshot();
      });
   });

   describe('Создание ссылки и кликов по ней', () => {
      afterEach(() => {
         document.body.innerHTML = '';
      });

      it('Создана ссылка "tensor" в body', () => {
         createLink();
         expect(document.body).toMatchSnapshot();
      });

      it('Первый клик по ссылке "tensor"', () => {
         createLink();
         document.body.getElementsByTagName('a')[0].click();
         expect(document.body).toMatchSnapshot();
      });

      it('Второй клик по ссылке "tensor"', () => {
         createLink();
         document.body.getElementsByTagName('a')[0].click();
         document.body.getElementsByTagName('a')[0].click();
         expect(document.location.href).toEqual("https://tensor.ru/");
      });
   });
});
