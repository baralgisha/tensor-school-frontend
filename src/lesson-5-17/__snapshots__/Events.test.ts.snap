// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Урок 5.17 - События Создание и удаление кнопки по клику Клик по кнопке "Удали меня" 1`] = `<body />`;

exports[`Урок 5.17 - События Создание и удаление кнопки по клику Создана кнопка в body "Удали меня" 1`] = `
<body>
  <button>
    Удали меня
  </button>
</body>
`;

exports[`Урок 5.17 - События Создание списка и запись атрибута по наведению Наведение мыши на элементы списка 1`] = `
<body>
  <ul>
    <li
      title="нулевой элемент"
    >
      нулевой элемент
    </li>
    <li
      title="первый элемент"
    >
      первый элемент
    </li>
    <li
      title="второй элемент"
    >
      второй элемент
    </li>
  </ul>
</body>
`;

exports[`Урок 5.17 - События Создание списка и запись атрибута по наведению Создан список в body из элементов массива 1`] = `
<body>
  <ul>
    <li>
      нулевой элемент
    </li>
    <li>
      первый элемент
    </li>
    <li>
      второй элемент
    </li>
  </ul>
</body>
`;

exports[`Урок 5.17 - События Создание ссылки и кликов по ней Первый клик по ссылке "tensor" 1`] = `
<body>
  <a
    href="https://tensor.ru/"
  >
    tensor https://tensor.ru/
  </a>
</body>
`;

exports[`Урок 5.17 - События Создание ссылки и кликов по ней Создана ссылка "tensor" в body 1`] = `
<body>
  <a
    href="https://tensor.ru/"
  >
    tensor
  </a>
</body>
`;
